js/modules/k6/experimental/fs/registry_test.go

```diff
@@ -0,0 +1,52 @@
+package fs
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFileRegistryOpen(t *testing.T) {
+	t.Parallel()
+
+	t.Run("open succeeds", func(t *testing.T) {
+		t.Parallel()
+
+		registry := &registry{}
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, "bonjour.txt", []byte("Bonjour, le monde"), 0o644)
+		})
+
+		_, gotBeforeOk := registry.files.Load("bonjour.txt")
+		gotData, gotErr := registry.open("bonjour.txt", fs)
+		_, gotAfterOk := registry.files.Load("bonjour.txt")
+
+		assert.False(t, gotBeforeOk)
+		assert.NoError(t, gotErr)
+		assert.Equal(t, []byte("Bonjour, le monde"), gotData)
+		assert.True(t, gotAfterOk)
+	})
+
+	t.Run("double open succeeds", func(t *testing.T) {
+		t.Parallel()
+
+		registry := &registry{}
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, "bonjour.txt", []byte("Bonjour, le monde"), 0o644)
+		})
+
+		firstData, firstErr := registry.open("bonjour.txt", fs)
+		_, gotFirstOk := registry.files.Load("bonjour.txt")
+		secondData, secondErr := registry.open("bonjour.txt", fs)
+		_, gotSecondOk := registry.files.Load("bonjour.txt")
+
+		assert.True(t, gotFirstOk)
+		assert.NoError(t, firstErr)
+		assert.Equal(t, []byte("Bonjour, le monde"), firstData)
+		assert.True(t, gotSecondOk)
+		assert.NoError(t, secondErr)
+		assert.Equal(t, firstData, secondData) // same pointer
```

Comment:
```suggestion
		assert.Same(t, firstData, secondData)
```
https://pkg.go.dev/github.com/stretchr/testify/assert#Same

Reply:
Actually, this is interesting, for some reason, the test fails with `Same`:

```
            	Error:      	Not same:
            	            	expected: 0x1400009ea00 []byte{0x42, 0x6f, 0x6e, 0x6a, 0x6f, 0x75, 0x72, 0x2c, 0x20, 0x6c, 0x65, 0x20, 0x6d, 0x6f, 0x6e, 0x64, 0x65}
            	            	actual  : 0x1400009ea00 []byte{0x42, 0x6f, 0x6e, 0x6a, 0x6f, 0x75, 0x72, 0x2c, 0x20, 0x6c, 0x65, 0x20, 0x6d, 0x6f, 0x6e, 0x64, 0x65}
```

But I fail to see why 😓 they look to be identical to me: same address, same content?

Reply:
After re-reading https://go.dev/blog/slices-intro I have a better intuition as to why `assert.Same` doesn't effectively test the two byte slices are the same here.

That's because, they are 🥁 not.

When the file registry returns the byte slice corresponding to the file's content, it returns a _copy_ of its internal slice, which points to the same underlying array `*ptr`, and has the same `len` and `cap` properties. Only the "slice header" is copied. But the result is that in that test, although `firstData`, and `secondData`'s internal pointer, len, and capacity are equal, their "slice header" address is different.

Apparently, `bytes.Equal` can help with that. I also read that checking if both slices' first byte has the same address would work too (as in they both point to the same first byte in memory).

Reply:
I've replaced the check with `assert.True(t, bytes.Equal(firstData, secondData))` 👍🏻